src/jit/gentree.cpp

```diff
@@ -8244,6 +8575,14 @@ GenTreePtr GenTree::GetChild(unsigned childNum)
     {
         return AsUnOp()->gtOp1;
     }
+    // Special case for assignment of dynamic block.
+    // This code is here to duplicate the former case where the size may be evaluated prior to the
+    // source and destination addresses. In order to do this, we treat the size as a child of the
+    // assignment.
+    if ((OperGet() == GT_ASG) && (gtOp.gtOp1->OperGet() == GT_DYN_BLK) && (childNum == 2))
+    {
+        return gtOp.gtOp1->AsDynBlk()->gtDynamicSize;
+    }
     else if (OperIsBinary())
```

Comment:
Yes - I don't think that we actually need to preserve this behavior long-term, but I was really trying to minimize diffs for this initial checkin. But there should be a TODO-1stClassStructs here to revisit this. I'll add one.

---
In reply to: [77259527](https://github.com/dotnet/coreclr/pull/7027#discussion_r77259527) [](ancestors = 77259527)